src/Client.php

```diff
@@ -0,0 +1,370 @@
+   * @throws SdkException On any other error
+   */
+  public function request(
+    string $method,
+    string $endpoint,
+    array $params = []
+  ) : Response {
+    try {
+      $params['headers'] =
+        ($params['headers'] ?? []) + $this->_getDefaultHeaders();
+
+      $response = new Response(
+        $this->_client->request($method, $endpoint, $params)
+      );
+      return $response;
+
+    } catch (ConnectException $e) {
+      throw new ApiException(ApiException::CANNOT_CONNECT, $e);
+    } catch (ClientException $e) {
+      switch ($e->getResponse()->getStatusCode()) {
+        case 401 :
+          $code = ApiException::UNAUTHORIZED;
+          break;
+        case 403:
+          $code = ApiException::FORBIDDEN;
+          break;
+        case 404:
+          $code = ApiException::NOT_FOUND;
+          break;
+        case 422:
+          $code = ApiException::UNPROCESSABLE_ENTITY;
+          break;
+        default:
+          $code = ApiException::BAD_REQUEST;
+          break;
+      }
+      throw new ApiException(
+        $code,
+        $e,
+        ['method' => $method, 'endpoint' => $endpoint]
+      );
+    } catch (ServerException $e) {
+      throw new ApiException(ApiException::SERVER_ERROR, $e);
+    } catch (RequestException $e) {
+      throw new ApiException(ApiException::REQUEST_FAILED, $e);
+    } catch (Throwable $e) {
+      throw new SdkException(SdkException::UNKNOWN_ERROR, $e);
+    }
+  }
+
+  /**
+   * Perform self-update.
+   *
+   * @return array Information about the update.
+   * @throws SdkException If update fails
+   */
+  public function selfUpdate() : array {
+    throw new SdkException(
+      SdkException::NOT_IMPLEMENTED,
+      ['class' => __CLASS__, 'method' => __FUNCTION__]
+    );
+  }
+
+  /**
+   * Does the SDK need to be updated?
+   *
+   * @return bool True if a newer SDK version is available; false otherwise
+   */
+  public function shouldUpdate() : bool {
+    throw new SdkException(
+      SdkException::NOT_IMPLEMENTED,
+      ['class' => __CLASS__, 'method' => __FUNCTION__]
+    );
+  }
+
+  /**
+   * Gets a log of API requests performed by this client.
+   *
+   * @return array[] Info about API request, categorized by endpoint
+   * @throws If request logging is disabled
+   */
+  public function getRequestLog() : array {
+    $config = $this->_config;
+    if (! ($config->get('debug') || $config->get('request.log'))) {
+      throw new SdkException(SdkException::REQUEST_LOG_NOT_ENABLED);
+    }
+
+    return $this->_request_log;
+  }
+
+  /**
+   * Gets default headers for API requests.
+   *
+   * @return array Map of http headers
+   */
+  protected function _getDefaultHeaders() : array {
+    $headers = [
+      'Accept' => 'application/json',
+      'Accept-language' => $this->_config->get('language'),
+      'Api-version' => static::API_VERSION,
+      'User-agent' => 'Nexcess-SDK/' . static::SDK_VERSION .
```

Comment:
This whole thing should be a class constant IMO. Also point out this is the PHP SDK.

Reply:
it can't be constant; it includes the guzzle UA (and i think it should)

will add the "php" part